### DISCUSSION TITLE
Can we have an option to bring back nvim-cmp instead of blink-cmp

### DISCUSSION THREAD

**Discussion by tony-sn** [LazyVim/LazyVim#5104] [Q&A] [answered]
Cannot deny the fact that blink-cmp works as its name, as speed of a blink, but it changes my work flow dramatically. Normally when my flow at typescript is Type > ESC > Insert Mode > Type > CR. It works normal at nvim-cmp.

The problem occurs as blink-cmp runs too quickly, now whenever I leave Insert Mode, close brackets ('>', for example, close a button tag) and press Enter, it already pick the first suggestion from blink-cmp to my code, therefore, I have to delete the suggestion as it is not intended. Or I have to include a SPACE after I finish closing the bracket.

I never experience this with nvim-cmp. Try to add nvim-cmp at Extras as suggested, but the cmp will be even worse, there is no suggestion at all. It would be great if we can have an option to keep using nvim-cmp.

<img width="824" alt="Screenshot 2024-12-16 at 10 38 37 am" src="https://github.com/user-attachments/assets/ab48bc3c-cc7a-4060-8eff-cefe8c790d8c" />

**Comment by dpetka2001** [comment, marked as answer]
I just tried `nvim-cmp` from LazyExtras and it works as expected. I don't do Typescript, so I just tried it with Lua and I get completions just fine. You must be doing something wrong. Here's a screencast with default LazyVim installation from scratch and enabling the `nvim-cmp` from `LazyExtras`.

[Screencast_2024-12-16-02-22-55.webm](https://github.com/user-attachments/assets/6d4763e2-b023-4e72-a1de-ae0a71629523)

**Reply by tony-sn** [reply]
Thanks, @dpetka2001 

I tried to enable it through:
```lua
require("lazy").setup({
    { import = "lazyvim.plugins.extras.coding.nvim-cmp" },
})
```

But my autocomplete table was not shown at all, enable it through your Lazy Extras buffer seems work perfectly now. Thanks so much. I am not ready to transition to blink.cmp yet.

**Reply by liocer** [reply]
Blink's been driving me nuts all morning. It doesn't seem to complete well at all for me at least. So going back to nvim-cmp is a relief

**Reply by tonypingco** [reply]
Agree, either blink is not ready or it's too quick that we need to adapt our flow because the auto suggestion process triggers that quick.

**Comment by horumyy** [comment]
It can be enabled back with `:LazyExtras` as stated [here](https://github.com/LazyVim/LazyVim/discussions/5032) (its for telescope and fzf but works to remove blink as well), since every update feels like screwing my workflow over and over I just figured out that changing `checker = { enabled = true }` to `false` on your `lazy.lua` will be a headache saver
